larpandoracontent/LArControlFlow/MasterAlgorithm.cc

```diff
@@ -509,14 +509,30 @@ StatusCode MasterAlgorithm::RunSlicing(const VolumeIdToHitListMap &volumeIdToHit
 
 StatusCode MasterAlgorithm::RunSliceReconstruction(SliceVector &sliceVector, SliceHypotheses &nuSliceHypotheses, SliceHypotheses &crSliceHypotheses) const
 {
+    SliceVector selectedSliceVector;
+    if (m_shouldRunSlicing && !m_sliceSelectionToolVector.empty())
+    {
+        SliceVector inputSliceVector(sliceVector);
+        for (SliceSelectionBaseTool *const pSliceSelectionTool : m_sliceSelectionToolVector)
+        {
+            pSliceSelectionTool->SelectSlices(this, inputSliceVector, selectedSliceVector);
+            inputSliceVector = selectedSliceVector;
+        }
+    }
+    else
+    {
+        selectedSliceVector = sliceVector;
```

Comment:
I suppose this copy means there's an overhead when running standard reconstruction, compared to before? Can't think of an immediate way to avoid this?

Reply:
I added move semantics to this case to avoid the copy.